### PR TITLE
[YANG] Add MUX_CABLE yang model 

### PR DESCRIPTION
**Pull request by zjswhhh**
#### Why I did it

Address issue https://github.com/sonic-net/sonic-buildimage/issues/10970 

sign-off: Jing Zhang zhangjing@microsoft.com

#### How I did it

Add `sonic-mux-cable.yang` and unit tests.

#### How to verify it
* Compile Compile target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl and target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl.
* Pass sonic-config-engine unit test. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### Link to config_db schema for YANG module changes

https://github.com/sonic-net/sonic-buildimage/blob/f8fe41a0238b8a7b9e32ae42262f41b63050c55f/src/sonic-yang-models/doc/Configuration.md#mux_cable

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Please link the GitHub issue you are trying to close in PR description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    container sonic-mux-cable {
+        
+        container MUX_CABLE {
+            
+            list MUX_CABLE_LIST {
+                
+                key "ifname";
+
+                leaf ifname {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:32]: Does mux cable have any dependency on interface?
For example, maybe we need to shutdown interface before configuring mux cable?

Comment (zjswhhh) [reply src/sonic-yang-models/yang-models/sonic-mux-cable.yang:32]: MUX_CABLE table is per interface. 

Why do we need to do that?

Comment (ganglyu) [reply src/sonic-yang-models/yang-models/sonic-mux-cable.yang:32]: You can use leafref, please refer to https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-lldp.yang#L111-L117

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf cable_type {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:38]: Please add description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    type enumeration {
+                        enum active-active;
+                        enum active-standby;
+                    }
+                    default active-standby;
+                }
+
+                leaf server_ipv4 {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:46]: Please add description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                    type inet:ipv4-prefix;
+                }
+
+                leaf server_ipv6 {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:50]: Please add description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                    type inet:ipv6-prefix;
+                }
+
+                leaf soc_ipv4 {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:54]: Please add description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                    type inet:ipv4-prefix;
+                }
+
+                leaf soc_ipv6 {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:58]: Please add description.

File: src/sonic-yang-models/yang-models/sonic-mux-cable.yang [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                    type inet:ipv6-prefix;
+                }
+
+                leaf state {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-mux-cable.yang:62]: Please add description.

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by zjswhhh** [comment]
**TASK#: 10997354**

**Pull request merged**